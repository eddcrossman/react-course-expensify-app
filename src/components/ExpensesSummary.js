import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'

require('numeral/locales/en-gb');
numeral.locale('en-gb');

export const ExpensesSummary = ( {expenseCount, expensesTotal} ) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
}

const mapStateToProps = (state)=> {
    const expenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: expenses.length,
        expensesTotal: expensesTotal(expenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);