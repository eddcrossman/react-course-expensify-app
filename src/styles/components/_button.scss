.button {
    background: $blue;
    border: none;
    color: white;
    border-bottom: 3px solid darken($blue, 10%);
    display: inline-block;
    text-decoration: none;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
}

.button--link {
    background: none;
    border: none;
}

.button--secondary {
    @extend .button;
    background: #888;
    border-bottom: 3px solid darken(#888888, 10%);

}