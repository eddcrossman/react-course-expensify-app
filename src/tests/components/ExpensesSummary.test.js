import React from 'react';
import { shallow }  from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('Should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={232} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with totals from expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={12232} />);
    expect(wrapper).toMatchSnapshot();
});