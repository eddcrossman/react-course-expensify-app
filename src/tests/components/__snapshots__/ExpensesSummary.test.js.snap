// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    £2.32
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with totals from expenses 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    £122.32
  </h1>
</div>
`;
